### PR TITLE
Jetpack connect: Separate from signup module

### LINKED ISSUE #12546: Signup: move `JetpackConnect` to a separate directory to reducess `signup` chunk size
We should move `client/signup/jetpack-connect` out of the `signup` chunk. It isn't required by any code inside `signup` itself.

This will slightly improve the loading time for new users.

### PR DESCRIPTION
**Pull request by sirreal**
Fixes #12546 

Move `jetpack-connect` into an independent module. I will revisit this PR after the following PRs are merged. This way lint errors can be fixed independently, since moving any files will provoke any lint errors in the affected files as if they were new errors.

Required PRs:

- [x] #13368 index (JetpackConnectMain)
- [x] #13369 sso
- [x] #13496 install-step
- [x] #13500 jetpack-connect-notices
- [x] #13504 plans-grid
- [x] #13638 plans-landing
- [x] #13639 site-url-input
- [x] authorize-form
   - [x] #13752 Separate/fix `SiteCard`
   - [x] #13753 Separate/fix `LoggedInForm` into own file
   - [x] #13798 Separate/fix `LoggedOutForm` into own file
   - [x] #14046 `AuthorizeForm` lint

~#13366~
~#13337~

### CODE REVIEW

File: client/jetpack-connect/jetpack-connect-notices.jsx [hunk 1]
```diff
@@ -35,81 +34,81 @@ export default React.createClass( {
...
 		if ( this.props.noticeType === 'wordpress.com' ) {
-			noticeValues.text = this.translate( 'Oops, that\'s us.' );
+			noticeValues.text = this.props.translate( 'Oops, that\'s us.' );
 			noticeValues.status = 'is-warning';
 			noticeValues.icon = 'status';
 			return noticeValues;
 		}
 		if ( this.props.noticeType === 'retryingAuth' ) {
-			noticeValues.text = this.translate( 'Error authorizing. Page is refreshing for an other attempt.' );
+			noticeValues.text = this.props.translate( 'Error authorizing. Page is refreshing for another attempt.' );
```
Comment (a8ci18n) [inline client/jetpack-connect/jetpack-connect-notices.jsx]: :ok: This change is safe, we'll keep the existing 17 translations.

File: client/jetpack-connect/index.jsx [hunk 2]
```diff
@@ -27,7 +29,7 @@ import versionCompare from 'lib/version-compare';
 import LocaleSuggestions from 'signup/locale-suggestions';
 import { recordTracksEvent } from 'state/analytics/actions';
 import MainWrapper from './main-wrapper';
-import StepHeader from '../step-header';
+import StepHeader from 'signup/step-header';
```
Comment (sirreal) [inline client/jetpack-connect/index.jsx]: Should `step-header` be moved to `client/components`?

File: client/jetpack-connect/index.jsx [hunk 3]
```diff
@@ -158,20 +155,21 @@ const JetpackConnectMain = React.createClass( {
 		}
 
 		if ( this.state.waitingForSites && ! this.props.isRequestingSites ) {
+			// eslint-disable-next-line react/no-did-update-set-state
```
Comment (sirreal) [inline client/jetpack-connect/index.jsx]: This usage seems safe for `setState` within `componentDidUpdate`, but proposals for a better solution are welcome.

**Comment by roccotripaldi** [comment]
I plan on reviewing / testing this more extensively, but what jumps out immediately is that this `jetpack-connect` should be in `blocks` folder, rather than a top-level `client` folder. @mtias what do you think?

**Comment by tyxla** [comment]
I'm leaning more towards segmenting this into several smaller PRs, some of them can be like this:
* Migrate this or that to Redux
* ES6ify this or that; fix linting errors.
* Do preparations for moving it in /blocks
* Actually moving `jetpack-connect` after all linting errors are fixed.

The problem with this large PR in that case is not only that it's difficult to review, but it's also very difficult to test. As we all know, JPC has a thousand of use cases and it will be much easier if we approach this task granularly and with smaller, more focused PRs. 

But generally, the smaller the PR, the easier to review and test, and the less chance to break stuff. And you'll avoid having to rebase a large PR a thousand times before it gets merged. 😉

**Comment by sirreal** [comment]
@tyxla Thanks, that's a good plan. The "meat" of this PR isn't really a big change, but the peripheral changes really inflate it. I'm closing this to open some new incremental PRs.

**Comment by sirreal** [comment]
This is ready to move forward, but I'll start with a clean slate in #14321

**Pull request closed without merging**